src/util/AllocatorWithLimit.h

```diff
@@ -173,32 +177,37 @@ class AllocatorWithLimit {
     const auto bytesNeeded = n * sizeof(T);
     const bool wasEnoughLeft =
         memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_return_false(
-            bytesNeeded);
+            MemorySize::bytes(bytesNeeded));
     if (!wasEnoughLeft) {
       clearOnAllocation_(n);
-      memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_throw(bytesNeeded);
+      memoryLeft_.ptr()->wlock()->decrease_if_enough_left_or_throw(
+          MemorySize::bytes(bytesNeeded));
     }
     // the actual allocation
     return allocator_.allocate(n);
   }
+  // Overload for use with `MemorySize`.
+  T* allocate(MemorySize n) { return allocate(n.getBytes()); }
```

Comment:
Do we need this? I think we only need the interface for a `std::allocator` and this overload can be removed without any loss.

Reply:
Funny thing: The current implementation of it is plain wrong. `allocate` doesn't take bytes, but the number of objects.

In light of this information, we definitely do not want any overloads of this kind.

Reply:
Deleted in [a2bfe49](https://github.com/ad-freiburg/qlever/pull/1112/commits/a2bfe498959ff1ab05e70901e5414f8d8a6419e5).